File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTPermissionDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTModuleDef } from "./TTModuleDef";
import { TTInterfaceDef } from "./TTInterfaceDef";
import { TTDefCollection } from "./TTDefCollection";
import { TTPermissionParameterDef } from "./TTPermissionParameterDef";

export class TTPermissionDef {
    public ModuleDefID: Guid;
    public ModuleDef: TTModuleDef;
    public ID: Guid;
    //public SecurityAuthority: TTSecurityAuthority;
    public PermissionDefID: Guid;
    public CodeName: string;
    public Name: string;
    public InterfaceDef: TTInterfaceDef;
    public InterfaceDefID: Guid;
    public BasePermissionDef: TTPermissionDef;
    public BasePermissionDefID: Guid;
    public Body: string;
    public ParameterDefs: TTDefCollection<TTPermissionParameterDef>;
    public AllParameterDefs: TTDefCollection<TTPermissionParameterDef>;
    //public RightDefs: TTDualKeyCollection<number, string, TTRightDef>;
    public ToString(): string {
        return null;
    }
}

File: AtlasModel/Enterprise/TTPermissionDef.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTPermissionDef
    {
        public Guid PermissionDefId { get; set; }
        public Guid? BasePermissionDefId { get; set; }
        public string Name { get; set; }
        public Guid? InterfaceDefId { get; set; }
        public string Body { get; set; }
        public Guid ModuleDefId { get; set; }
        public string CheckOutId { get; set; }
        public short CheckOutStatus { get; set; }
    }
}